lib/core/test/unit/Cardano/Wallet/Api/TypesSpec.hs

```diff
@@ -722,6 +733,20 @@ instance Arbitrary (PostTransactionFeeData t) where
     arbitrary = genericArbitrary
     shrink = genericShrink
 
+instance Arbitrary PostExternalTransactionData where
+    arbitrary = genericArbitrary
+    shrink = genericShrink
+
+instance Arbitrary ByteString where
+    shrink bytes | BS.null bytes = []
+    shrink bytes =
+        [ BS.take (BS.length bytes `div` 2) bytes
+        , BS.drop 1 bytes
+        ]
+    arbitrary = do
+        count <- choose (0, 32)
+        BS.pack <$> replicateM count arbitrary
```

Comment:
I think we should probably find a way to avoid defining an orphan `Arbitrary` instance for `ByteString` here, and instead place this logic within the `PostExternalTransactionData` `Arbitrary` instance.

Reply:
Or do like QuickCheck does and make a `newtype ShortByteString = ShortByteString { getShortByteString :: ByteString }`.

Reply:
indeed! Eliminated ByteString instance by placing the logic inside  `PostExternalTransactionData` instance